stubs/redis/redis/client.pyi

```diff
@@ -729,9 +729,10 @@ class Redis(Generic[_StrType]):
     def hincrbyfloat(self, name: _Key, key: _Key, amount: float = ...) -> float: ...
     def hkeys(self, name: _Key) -> List[_StrType]: ...
     def hlen(self, name: _Key) -> int: ...
-    def hset(
-        self, name: _Key, key: Optional[_Key], value: Optional[_Value], mapping: Optional[Mapping[_Value, _Value]] = ...
-    ) -> int: ...
+    @overload
+    def hset(self, name: _Key, key: _Key, value: _Value, mapping: Optional[Mapping[_Value, _Value]] = ...) -> int: ...
+    @overload
+    def hset(self, name: _Key, *, mapping: Mapping[_Value, _Value]) -> int: ...
```

Comment:
It is possible to do `.hset(name, key, None, mapping)`, so it's tempting to do this:

```python
    def hset(self, name: _Key, key: _Key, value: None = ..., mapping: Mapping[_Value, _Value]) -> int: ...
```

But this isn't possible, because arguments without default value must be first (unless they are keyword-only). So instead, we need two overloads:

```suggestion
    @overload
    def hset(self, name: _Key, key: _Key, value: None, mapping: Mapping[_Value, _Value]) -> int: ...
    @overload
    def hset(self, name: _Key, key: _Key, *, value: None = ..., mapping: Mapping[_Value, _Value]) -> int: ...
```

Reply:
I don' think `hset(name, None, None, mapping)` is valid. From the redis-py implementation, it adds a k-v pair with key=None (null) and value=None (null), and adds a mapping of k-v pairs alongside. For both of redis-py and redis, neither key or value should be None. It will cause runtime exception.

Reply:
The overloads I am suggesting don't allow `hset(name, key, None, mapping)` with a non-None `key`, but they allow `hset(name, None, None, mapping)`.  For the latter, [this line](https://github.com/andymccurdy/redis-py/blob/1a41cfd95a53a95b078084d8627be6b6fba3bb71/redis/client.py#L3146) will ignore the two `None`s.

Reply:
Oh, and the mapping has to be non-empty for this to work, although there's no way to enforce that with a type hint.

Reply:
> Oh, and the mapping has to be non-empty for this to work, although there's no way to enforce that with a type hint.

Yes that's a limitation

Reply:
>I see what you mean. Does this work?

Yes, with one small limitation: it doesn't accept `hset(name, key=None, value=None, mapping=mapping)` even though it works at runtime. However, I will accept that too if you prefer that, since this is unlikely to be a problem in practice.

>Yes that's a limitation. Do you ignore it?

Like I said, we can't do anything about it, so we have to ignore it.